### ISSUE TITLE
EvolvePokemon task does not work properly

### ISSUE THREAD

**Issue by Pawamoy** [PokemonGoF/PokemonGo-Bot#2501]
### Expected Behavior
Follow instructions in config: evolve all possible pokemons, or only specified ones

### Actual Behavior
Stop after some evolutions. Does not even start to evolve if "evolve_all" is something else than "all"

### Steps to Reproduce
(dev branch)
I first launched the bot with

```json
"tasks": [
      {
        "type": "EvolvePokemon",
        "config": {
          "evolve_all": "Pidgey,Rattata,Weedle,Caterpie",
          "evolve_cp_min": 15,
          "evolve_speed": 15,
          "use_lucky_egg": false
        }
      }
    ],
```

Seeing that it was doin' nothing, I set evolve_all to "all". Then it starts by evolving magicarp and dratini :angry: After that it evolves something like 5 Pidgeys, 2 Rattatas and then it just stops !
I launch and relaunch it several times, changing evolve_cp_min, resetting evolve_speed to 20 but no success.

### Other Information
OS: Ubuntu 16
Git Commit: 361c0fa11783085ea201a71a99f6c43859e8c239
Python Version: 2.7.12

**Comment by Pawamoy** [comment]
> [17:43:42] Using lucky egg ... you have 1
[17:43:42]  Successfully used lucky egg... (0 left!) 
[17:43:44] Successfully evolved Rattata with 339 CP and 0.82 IV!
[17:43:56] Successfully evolved Rattata with 222 CP and 0.89 IV!
[17:44:07] Successfully evolved Pidgey with 370 CP and 0.87 IV!
[17:44:22] Successfully evolved Pidgey with 365 CP and 0.82 IV!
[17:44:35] Successfully evolved Weedle with 208 CP and 0.89 IV!
[17:44:49] Evolved 5 pokemon!

I  had something like 50 Pidgeys with 1000+ candies...

**Comment by sffetlio** [comment]
Your config might be outdated.
Try something like:
```
{
        "type": "EvolvePokemon",
        "config": {
            "evolve_all": "Pidgey,Rattata,Weedle,Caterpie",
            "first_evolve_by": "cp",
            "evolve_above_cp": 500,
            "evolve_above_iv": 0.8,
            "logic": "or",
            "evolve_speed": 20,
            "use_lucky_egg": false
        }
},
```

**Comment by Pawamoy** [comment]
Okay it worked, sorry for the ticket, hard to follow dev changes :)
